### ISSUE TITLE
Why use json_file?

### ISSUE THREAD

**Issue by ioquatix** [lemurheavy/coveralls-public#1442]
> Because a project with all it’s coverage and source can be large, you must write your API request to a JSON file and send it with the form parameter json_file. You can optionally GZIP your json – set the content-type header to “gzip/json” in that case.

I don't understand how this makes any difference.

Can't you just set your transfer-encoding to gzip and send the json as the post data?

**Comment by ioquatix** [comment]
This is probably still an issue?

**Comment by afinetooth** [comment]
Hi @ioquatix, Thanks for the feedback. Because this is more of a design discussion than a specific user issue, we'd prefer to close it and have you or other users open a new issue so we can focus on a specific diagnosis and solution. 

Are you having a specific problem submitting a POST to https://coveralls.io/api/v1/jobs?

If so, it will help if you can share your CI build log. Even better if you can re-run the build in [verbose mode](https://docs.coveralls.io/troubleshooting#put-it-in-verbose--v) and share the debug output with us.

Since you seem to be using the [coveralls-ruby](https://github.com/lemurheavy/coveralls-ruby) integration, you can enable [verbose mode](https://docs.coveralls.io/troubleshooting#put-it-in-verbose--v) by adding the following CI environment variable:
`COVERALLS_NOISY=true`

That will show the exact JSON you're posting to the [Coveralls API ](https://docs.coveralls.io/api-reference) so we can examine it.

If your repo is private or your build log contains sensitive information, you can send it to us at support@coveralls.io.

**Comment by ioquatix** [comment]
I think it's more that your API for submitting the coverage data could be simpler - prefer to use the standard post body and gzip transfer encoding.

**Comment by ioquatix** [comment]
As outlined in https://github.com/lemurheavy/coveralls-public/issues/1442#issuecomment-459615076 your implementation seems incredibly round-a-bout.

**Comment by afinetooth** [comment]
Hi @ioquatix. Got it. I've submitted a request for more background on that choice and have added your recommendation to our backlog. I'll feed back with any responses. Sorry I'm not able to address it myself at this time.

I recall you've implemented [this coveralls integration](https://github.com/ioquatix/covered) which I promised to add to our list of integrations under Ruby. I'm afraid it's still waiting on a documentation update that will reorganize that particular section. I will do my best to find an interim solution to listing it that so other Ruby users can check it out. Thanks.

**Comment by afinetooth** [comment]
@ioquatix update: Doc update adding [covered](https://github.com/ioquatix/covered) to our Ruby/Rails integrations has been submitted as a PR. Likely to be published this week or early next week.

**Comment by ioquatix** [comment]
Awesome, thanks!

**Comment by ioquatix** [comment]
It would be pretty awesome to revisit our coveralls support and make sure it's the best it can be.
